src/Core/Services/MetadataProviders/DatabaseObjectConverter.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.DatabasePrimitives;
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders
+{
+    /// <summary>
+    /// This is a converter to serialize and deserialize the DatabaseObject
+    /// this adds a typename field at the object level, this is used when you deserialize the different child objects : DatabaseTable,
+    /// DatabaseView as DatabaseObject is an abstract class
+    /// This is also required as there is no TypeNameHandling support in System.text.Json and we need to explicitly add a TypeName field to distinguish
+    /// </summary>
+    public class DatabaseObjectConverter : JsonConverter<DatabaseObject>
+    {
+        public override DatabaseObject Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            using (JsonDocument document = JsonDocument.ParseValue(ref reader))
+            {
+                JsonElement root = document.RootElement;
+                string typeName = root.GetProperty("TypeName").GetString() ?? throw new JsonException("TypeName is missing");
+                Type concreteType = GetTypeFromName(typeName);
+
+                if (concreteType == null)
+                {
+                    throw new JsonException($"Unknown type: {typeName}");
+                }
+
+                DatabaseObject objA = (DatabaseObject)JsonSerializer.Deserialize(document, concreteType, options)!;
+
+                return objA;
+            }
+        }
+
+        public override void Write(Utf8JsonWriter writer, DatabaseObject? value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+
+            // Add TypeName
+            writer.WriteString("TypeName", value!.GetType().AssemblyQualifiedName);
```

Comment:
nit: I see TypeName being used in multiple places in this class. can we move it to a constant?